### ISSUE TITLE
Add no-else-assert check

### ISSUE THREAD

**Issue by EugeneZelenko** [pylint-dev/pylint#9347]
### Current problem

It'll be great to have `no-else-assert` check in addition to existing `no-else-*` checks.

### Desired solution

It'll be great to have `no-else-assert` check in addition to existing `no-else-*` checks.

**Comment by Pierre-Sassoulas** [comment]
Hey, @EugeneZelenko thank you for opening the issue. Could you edit your issue with some functional test case where the message should be raised (or not), please ?

**Comment by EugeneZelenko** [comment]
Something like:

```
def function(arg):
    if isinstance(arg, str):
        print('string')
    elif isinstance(arg, list):
        print('list')
    else:
        assert False # Unexpected behavior
```

**Comment by jacobtylerwalls** [comment]
What's the case for considering this a problem?

**Comment by Pierre-Sassoulas** [comment]
Your code example doesn't look similar to the ``no-else-x`` checks. Did you mean this ?
```python
def function(arg):
    if isinstance(arg, str):
        return "string"
    else: # R1705: Unnecessary "else" after "return" (already raised)
        assert False
```
This already raises something, so did you mean this ?
```python
def function(arg):
    if isinstance(arg, str):
        assert False  # Unexpected behavior
    else: # This should start raising no-else-assert
        return "int"
```

**Comment by jacobtylerwalls** [comment]
I'm not in favor, as this would encourage treating asserts as control flow switches, which isn't true if you're running python in `-O` mode and and even if not, confuses the difference between assertions and exceptions.
